tests/page/expect-to-have-value.spec.ts

```diff
@@ -45,6 +45,13 @@ test('should support failure', async ({ page }) => {
   expect(stripAnsi(error.message)).toContain('"Text content"');
 });
 
+test('should normalize whitespace', async ({ page }) => {
+  await page.setContent('<input id=node></input>');
+  const locator = page.locator('#node');
+  await locator.fill('foo&nbsp;bar\nbaz');
```

Comment:
This results in exactly this text being inserted into the field, which is different from non-breaking whitespaces rendered as whitespace but returned as `&nbsp;` for the input value. Reading the bug I see that we were not able to find a good repro. Let's start with composing a good test case that fails with the current API before implementing this feature.

![image](https://github.com/user-attachments/assets/f7a5d310-0176-4481-901c-c0eb1a349695)